src/main/java/org/intellij/sdk/codesync/CodeSyncLogger.java

```diff
@@ -0,0 +1,119 @@
+            CloudWatchLogsClient logsClient = CloudWatchLogsClient.builder()
+                    .region(Region.US_EAST_1)
+                    .credentialsProvider(() -> AwsBasicCredentials.create(accessKey, secretKey))
+                    .build();
+            DescribeLogStreamsRequest logStreamRequest = DescribeLogStreamsRequest.builder()
+                    .logGroupName(logGroupName)
+                    .logStreamNamePrefix(streamName)
+                    .build();
+            DescribeLogStreamsResponse describeLogStreamsResponse = logsClient.describeLogStreams(logStreamRequest);
+
+            // Assume that a single stream is returned since a specific stream name was specified in the previous request.
+            String sequenceToken = describeLogStreamsResponse.logStreams().get(0).uploadSequenceToken();
+
+            // Build an input log message to put to CloudWatch.
+            JSONObject msg = new JSONObject();
+            msg.put("msg", message);
+            msg.put("source", DIFF_SOURCE);
+            InputLogEvent inputLogEvent = InputLogEvent.builder()
+                    .message(msg.toJSONString())
+                    .timestamp(System.currentTimeMillis())
+                    .build();
+
+            // Specify the request parameters.
+            // Sequence token is required so that the log can be written to the
+            // latest location in the stream.
+            PutLogEventsRequest putLogEventsRequest = PutLogEventsRequest.builder()
+                    .logEvents(Collections.singletonList(inputLogEvent))
+                    .logGroupName(logGroupName)
+                    .logStreamName(streamName)
+                    .sequenceToken(sequenceToken)
+                    .build();
+
+            logsClient.putLogEvents(putLogEventsRequest);
+
+            System.out.println("Successfully put CloudWatch log event");
+        } catch (CloudWatchException e) {
+            e.printStackTrace();
+            throw e;
+        }
+    }
+
+    public static void logEvent(String message, String userEmail) {
+        LOGGER.log(Level.SEVERE, message);
+        UserFile.User user = null;
+        UserFile userFile = null;
+
+        try {
+            userFile = new UserFile(USER_FILE_PATH);
+        } catch (FileNotFoundException | InvalidYmlFileError e) {
+            e.printStackTrace();
+        }
+
+        if (userEmail != null && userFile != null) {
+            user = userFile.getUser(userEmail);
+        } else if (userFile != null) {
+            user = userFile.getUser();
+        }
+
+//        SequenceTokenFile sequenceTokenFile;
+        if (user == null) {
+            // Can't log any thing.
+            return;
+        } else if (user.getAccessKey() == null || user.getSecretKey() == null || user.getUserEmail() == null) {
+            // can't log anything
+            return;
+        }
+        // I do not think we need a separate sequence token file in java. Need to verify it though.
```

Comment:
What do you mean by separte sequence token file?

Reply:
python code creates `sequence_token.yml` file to store the sequence token. Java client seems to handles sequencing on its own and hence we do not need that file in the plugin.

Reply:
Wow, amzing. Wondering if there is something similar in Python & JS side, both handle token thing manully.

Reply:
Still we need to save latest sequence token in the file so that other plugins get the correct token. Kindly make sure it is updated in the file during the process